stdlib/universe/map_vectorized.go

```diff
@@ -0,0 +1,250 @@
+package universe
+
+import (
+	"context"
+
+	"github.com/apache/arrow/go/v7/arrow/memory"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/compiler"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+const (
+	vectorizedMapKind = "vectorizedMap"
+)
+
+func init() {
+	plan.RegisterPhysicalRules(vectorizeMapRule{})
+	execute.RegisterTransformation(vectorizedMapKind, createVectorizedMapTransformation)
+}
+
+type vectorizedMapProcedureSpec struct {
+	plan.DefaultCost
+	Fn interpreter.ResolvedFunction
+}
+
+func (v *vectorizedMapProcedureSpec) Kind() plan.ProcedureKind {
+	return vectorizedMapKind
+}
+
+func (v *vectorizedMapProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(vectorizedMapProcedureSpec)
+	*ns = *v
+	ns.Fn = v.Fn.Copy()
+	return ns
+}
+
+type vectorizeMapRule struct{}
+
+func (v vectorizeMapRule) Name() string {
+	return "vectorizeMapRule"
+}
+
+func (v vectorizeMapRule) Pattern() plan.Pattern {
+	return plan.MultiSuccessor(MapKind)
+}
+
+func (v vectorizeMapRule) Rewrite(ctx context.Context, node plan.Node) (plan.Node, bool, error) {
+	mapSpec := node.ProcedureSpec().(*MapProcedureSpec)
+	if mapSpec.Fn.Fn.Vectorized == nil {
+		return node, false, nil
+	}
+
+	return plan.ReplacePhysicalNodes(ctx, node, node, vectorizedMapKind, &vectorizedMapProcedureSpec{
+		Fn: interpreter.ResolvedFunction{
+			Fn:    mapSpec.Fn.Fn.Vectorized,
+			Scope: mapSpec.Fn.Scope,
+		},
+	}), true, nil
+}
+
+func createVectorizedMapTransformation(
+	id execute.DatasetID,
+	mode execute.AccumulationMode,
+	spec plan.ProcedureSpec,
+	a execute.Administration,
+) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*vectorizedMapProcedureSpec)
+	if !ok {
+		return nil, nil, errors.Newf(codes.Internal, "invalid spec type %T", spec)
+	}
+
+	tr := &mapTransformation{
+		ctx: a.Context(),
+		fn: &mapVectorFunc{
+			fn: execute.NewVectorMapFn(s.Fn.Fn, compiler.ToScope(s.Fn.Scope)),
+		},
+	}
+	return execute.NewGroupTransformation(id, tr, a.Allocator())
+}
+
+type mapVectorFunc struct {
```

Comment:
Somehow Git lost track of it, but I meant to just change the name `map2.go` to `map_vectorized.go`. All of the code for the rest of the file was in `map2.go`, and I didn't need to change it.